source/Messages.cpp

```diff
@@ -15,15 +15,22 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include "Messages.h"
 
+#include "Color.h"
+#include "GameData.h"
+
+#include <cassert>
 #include <mutex>
 
 using namespace std;
 
 namespace {
+	const int MAX_LOG = 10000;
```

Comment:
is there a specific reason for 10000?
It seems like a lot, imagine scrolling through 10000 messages, wouldn't a hundred be enough?